File: GameServer/GameServerApp/GameServerApp/Proto/CreateProto/GameServerPageRequestProto.cs
//===================================================
//作    者：xxx
//创建时间：2019-05-18 15:13:23
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary> 请求服务器组列表(请求) </summary>
public struct GameServerPageRequestProto : IProto
{
    public ushort ProtoCode { get { return 10111; } }


    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            return ms.ToArray();
        }
    }

    public static GameServerPageRequestProto GetProto(byte[] buffer)
    {
        GameServerPageRequestProto proto = new GameServerPageRequestProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
        }
        return proto;
    }
}

File: GameServer/GameServerApp/GameServerApp/Proto/CreateLuaProto/GameServerPageRequestProto.lua
--请求服务器组列表(请求)
GameServerPageRequestProto = { ProtoCode = 10111 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
GameServerPageRequestProto.__index = GameServerPageRequestProto;

function GameServerPageRequestProto.New()
    local self = { }; --初始化self
    setmetatable(self, GameServerPageRequestProto); --将self的元表设定为Class
    return self;
end


--发送协议
function GameServerPageRequestProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);


    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function GameServerPageRequestProto.GetProto(buffer)

    local proto = GameServerPageRequestProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);


    ms:Dispose();
    return proto;
end